compiler/lib/src/main/scala/codegen/CppWriter/FormatCppWriter.scala

```diff
@@ -9,38 +9,37 @@ import fpp.compiler.util._
 object FormatCppWriter {
 
   /** Get the PRI macro for a decimal integer type */
-  def getDecimalFormat(name: Ast.TypeInt): String =
-    (TypeUtils.signedness(name), TypeUtils.width(name)) match {
-      case (TypeUtils.Unsigned, w) => s"%\" PRIu${w.toString} \""
-      case (TypeUtils.Signed, w) => s"%\" PRIi${w.toString} \""
+  def getDecimalFormat(pit: Type.PrimitiveInt): String =
+    (pit.signedness, pit.bitWidth) match {
+      case (Type.PrimitiveInt.Unsigned, w) => s"%\" PRIu${w.toString} \""
+      case (Type.PrimitiveInt.Signed, w) => s"%\" PRIi${w.toString} \""
     }
 
   /** Write an FPP format field as a C++ format string */
-  def writeField(f: Format.Field, tn: AstNode[Ast.TypeName]): String = {
+  def writeField(s: CppWriterState, f: Format.Field, tn: AstNode[Ast.TypeName]): String = {
```

Comment:
Do we need the type name if we are using the type directly from the type map? Maybe we could pass in the field and type instead of state, field, and type name? Just a thought but it also looks good the way it is here!

Reply:
If we did that, then we would have to look up the type every time we called this function. I don't have a strong opinion, but I think it's slightly better to pass in the raw data and look up the type in this function. In particular, now that we have introduced the notion of an "underlying type," looking up the type can be a source of bugs. I'm wondering what other bugs like this are still in the code.